include/revng/ABI/Definition.h

```diff
@@ -310,6 +311,27 @@ class Definition : public generated::Definition {
   bool
   checkCompatibility(const SortedVector<ArgReg> &ArgumentRegisters,
                      const SortedVector<RVReg> &ReturnValueRegisters) const;
+
+  /// \brief Compute the natural alignment of the type in accordance with
+  /// the current ABI
+  ///
+  /// \note  It mirrors, `model::Type::size()` pretty closely, see documentation
+  ///        related to it (and usage of the coroutines inside this codebase
+  ///        in general) for more details on how it works.
```

Comment:
I've some FUD about this... can we have a test?
Maybe even just a unit test, testing our expectations in non-trivial cases.

Reply:
Sounds good.

P. S. I definitely share your FUD, that's exactly why I'm so adamant on in being a 100% mirror of well tested by real usage `size()`, to the point I'd rather also change `size()` (and fix a bunch of stuff everywhere because of it) than let them go out of sync.

Reply:
(REMINDER to myself to leave a comment somewhere near the definition of `size()` so that hopefully anyone changing it won't forget about its little brother as well)

Reply:
Please review: `tests/unit/Alignment.cpp`.

(remainder is done)

Reply:
I'm closing this in favour of the newer thread: https://github.com/revng/revng/pull/232#discussion_r1121326768